profiler_test.go

```diff
@@ -0,0 +1,396 @@
+package sentry
+
+import (
+	"os"
+	"runtime"
+	"strconv"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+// Test ticker that ticks on demand instead of relying on go runtime timing.
+type profilerTestTicker struct {
+	c chan time.Time
+}
+
+func (t *profilerTestTicker) Channel() <-chan time.Time {
+	return t.c
+}
+
+func (t *profilerTestTicker) Stop() {}
+
+func (t *profilerTestTicker) Tick() {
+	t.c <- time.Now()
+	time.Sleep(time.Millisecond) // Allow the goroutine to pick up the tick from the channel.
+}
+
+func setupProfilerTestTicker() *profilerTestTicker {
+	ticker := &profilerTestTicker{c: make(chan time.Time, 1)}
+	profilerTickerFactory = func(d time.Duration) profilerTicker { return ticker }
+	return ticker
+}
+
+func restoreProfilerTicker() {
+	profilerTickerFactory = profilerTickerFactoryDefault
+}
+
+func TestProfilerCollection(t *testing.T) {
+	if !isCI() {
```

Comment:
Could you clarify what kind of flakiness is seen in CI (here, and/or in other places)?

Not a huge fan of having a lot of `isCI` checks, so e.g. if it can be remedied by having a test that runs a few hundred milliseconds longer -- I'd prefer that.

Reply:
It was extremely flaky, depending on the machine load. Go time ticker only says it will call your code _some time_ after the given period. Doesn't really guarantee anything (which is understandable, under load it couldn't really and it depends on what other routines are doing, whether they can be parked or not). 
Therefore, I've added test cases with manual ticker (below this one) but also kept the standard one to run at least locally so we have some chance of knowing. I can prolong the test case run to seconds if running in CI and we can see how it goes. I'd prefer doing so in another PR though so that it can be more easily reverted if it breaks on the main branch.

Reply:
> I'd prefer doing so in another PR

Yes, sounds good 👍